File: programmers/0911_kakao/q2.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        StringBuilder sb = new StringBuilder();
        
        while(n != 0) {
            int r = n % k;
            n /= k;
            sb.append(r);
        }
        
        String str = sb.reverse().toString();
        int len = str.length();
        ArrayList <Long> list = new ArrayList<> ();
        long num = 0;
        int max = 0;
        
        for(int i = 0; i < len; i++) {
            if(str.charAt(i) == '0') {
                if(num != (long) 0) {
                    list.add(num);
                    
                    if(max < Math.sqrt(num))
                        max = ((int) Math.sqrt(num)) + 1;
                }
                num = 0;
            } else {
                num *= 10;
                num += (long) (str.charAt(i) - '0');
            }
        }
        
        if(num != (long) 0) {
            list.add(num);
            
            if(max < Math.sqrt(num))
                max = ((int) Math.sqrt(num)) + 1;
        }
        
        boolean isPrime[] = new boolean[max + 1];
        
        for(int i = 2; i <= max; i++)
            isPrime[i] = true;
        
        for(int i = 2; i <= max; i++)
            for(int j = i + i; j <= max; j += i)
                isPrime[j] = false;
        
        ArrayList <Integer> primeList = new ArrayList<>();
        
        for(int i = 2; i <= max; i++)
            if(isPrime[i])
                primeList.add(i);
        
        for(int i = 0; i < list.size(); i++) {
            num = list.get(i);
            int idx = 0;
            boolean flag = true;
            
            if(num == (long) 1)
                flag = false;
            
            while(idx < primeList.size() && primeList.get(idx) < num) {
                if(num % primeList.get(idx) == 0) {
                    flag = false;
                    break;
                }
                idx++;
            }
            
            if(flag)
                answer++;
        }
        
        return answer;
    }
}

File: programmers/0911_kakao/q2.py
def is_prime(num):
    if num == 2:
        return True
    for i in range(2, int(num**1/2) + 1):
        if num%i == 0:
            return False
    return True

def convert(num, base) :
    q, r = divmod(num, base)
    if q == 0 :
        return str(r)
    else :
        return convert(q, base) + str(r)

def solution(n, k):
    num_list = []
    for s in convert(n, k).split('0'):
        if len(s) != 0:
            num_list.append(int(s))
    answer = 0
    for num in num_list:
        if num > 1 and is_prime(num):
            answer += 1
    return answer

n = 123456
k = 10
print(solution(n, k))